client/my-sites/earn/components/add-edit-coupon-modal/test/products-selector.jsx

```diff
@@ -0,0 +1,183 @@
+	sites: { items: {} },
+	siteSettings: { items: {} },
+	ui: { selectedSiteId: 1 },
+	memberships: {
+		couponList: {
+			items: {},
+		},
+		productList: {
+			items: {
+				1: [ testProduct1, testProduct2 ],
+			},
+		},
+	},
+};
+
+function WrappedProductsSelector( props ) {
+	const mockStore = configureStore();
+	const store = mockStore( initialState );
+
+	return (
+		<Provider store={ store }>
+			<ProductsSelector { ...props } />
+		</Provider>
+	);
+}
+
+describe( 'ProductsSelector', () => {
+	beforeEach( () => {
+		jest.clearAllMocks();
+
+		useTranslate.mockImplementation( mockUseTranslate );
+		translate.mockImplementation( mockTranslate );
+		__.mockImplementation( mockTranslate );
+	} );
+
+	test( 'should render with any product selected when no specific products have been selected', () => {
+		let editedPlanIdsAllowList = [];
+		const setEditedPlanIdsAllowList = ( list ) => {
+			editedPlanIdsAllowList = list;
+		};
+		const { container } = render(
+			<WrappedProductsSelector
+				onSelectedPlanIdsChange={ ( list ) => setEditedPlanIdsAllowList( list ) }
+				initialSelectedList={ editedPlanIdsAllowList }
+				allowMultiple={ true }
+			/>
+		);
+		expect( container ).toMatchSnapshot();
+	} );
+
+	test( 'should render with 1 product selected when 1 product has been selected', () => {
+		let editedPlanIdsAllowList = [ testProduct1.ID ];
+		const setEditedPlanIdsAllowList = ( list ) => {
+			editedPlanIdsAllowList = list;
+		};
+		const { container } = render(
+			<WrappedProductsSelector
+				onSelectedPlanIdsChange={ ( list ) => setEditedPlanIdsAllowList( list ) }
+				initialSelectedList={ editedPlanIdsAllowList }
+				allowMultiple={ true }
+			/>
+		);
+		expect( container ).toMatchSnapshot();
+	} );
+
+	test( 'should render with 2 products selected when 2 products have been selected', () => {
+		let editedPlanIdsAllowList = [ testProduct1.ID, testProduct2.ID ];
+		const setEditedPlanIdsAllowList = ( list ) => {
+			editedPlanIdsAllowList = list;
+		};
+		const { container } = render(
+			<WrappedProductsSelector
+				onSelectedPlanIdsChange={ ( list ) => setEditedPlanIdsAllowList( list ) }
+				initialSelectedList={ editedPlanIdsAllowList }
+				allowMultiple={ true }
+			/>
+		);
+		expect( container ).toMatchSnapshot();
+	} );
+
+	test( 'clicking the button should show a dropdown with selected items selected', () => {
+		let editedPlanIdsAllowList = [ testProduct1.ID, testProduct2.ID ];
+		const setEditedPlanIdsAllowList = ( list ) => {
+			editedPlanIdsAllowList = list;
+		};
+		render(
+			<WrappedProductsSelector
+				onSelectedPlanIdsChange={ ( list ) => setEditedPlanIdsAllowList( list ) }
+				initialSelectedList={ editedPlanIdsAllowList }
+				allowMultiple={ true }
+			/>
+		);
+
+		expect( document.body ).toMatchSnapshot();
+
+		act( () => screen.getByRole( 'button', { name: '2 products selected.' } ).click() );
+
+		expect( screen.getByRole( 'menuitemcheckbox', { name: 'Any product' } ) ).not.toBeChecked();
```

Comment:
Nit: I'd assign the DOM element getter calls to variables to improve readability. Something like:

```
const anyProductCheckbox = screen.getByRole( 'menuitemcheckbox', { name: 'Any product' } );

expect( anyProductCheckbox ).not.toBeChecked();
```

Which makes it easier to read (like plain English).